core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp

```diff
@@ -69,14 +69,15 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::Cuda> {
   Policy const& get_policy() const { return m_policy; }
 
   inline __device__ void operator()() const {
-    const Member work_stride = blockDim.y * gridDim.x;
-    const Member work_end    = m_policy.end();
+    const auto work_stride = Member(blockDim.y) * gridDim.x;
```

Comment:
Are we not running clang-tidy on this?
I expected `bugprone-implicit-widening-of-multiplication-result` can detect this?!

Reply:
I don't know if `clang-tidy` detects anything in device code.

Reply:
https://gitlab.spack.io/kokkos/kokkos/-/jobs/15256352 has
```
/builds/kokkos/kokkos/core/src/HIP/Kokkos_HIP_ParallelFor_Range.hpp:67:24: error: comparison of integers of different signs: 'Member' (aka 'int') and 'unsigned int' [-Werror,-Wsign-compare]
   67 |          iwork = iwork < work_end - work_stride ? iwork + work_stride
      |                  ~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~
```